front/pages/api/w/[wId]/assistant/skill_configurations/[sId]/index.ts

```diff
@@ -0,0 +1,293 @@
+  name: t.string,
+  description: t.string,
+  instructions: t.string,
+  tools: t.array(
+    t.type({
+      mcpServerViewId: t.string,
+    })
+  ),
+});
+
+type PatchSkillConfigurationRequestBody = t.TypeOf<
+  typeof PatchSkillConfigurationRequestBodySchema
+>;
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorResponse<
+      | GetSkillConfigurationResponseBody
+      | PatchSkillConfigurationResponseBody
+      | DeleteSkillConfigurationResponseBody
+    >
+  >,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.getNonNullableWorkspace();
+
+  const featureFlags = await getFeatureFlags(owner);
+  if (!featureFlags.includes("skills")) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "app_auth_error",
+        message: "Skill builder is not enabled for this workspace.",
+      },
+    });
+  }
+
+  const sId = req.query.sId as string;
+  const skillResource = await SkillConfigurationResource.fetchBySId(auth, sId);
+
+  if (!skillResource) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "skill_not_found",
+        message: "The skill you're trying to access was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET": {
+      const skillConfiguration = skillResource.toJSON();
+      return res.status(200).json({ skillConfiguration });
+    }
+
+    case "PATCH": {
+      const bodyValidation = PatchSkillConfigurationRequestBodySchema.decode(
+        req.body
+      );
+
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${pathError}`,
+          },
+        });
+      }
+
+      const body: PatchSkillConfigurationRequestBody = bodyValidation.right;
+
+      // Check if user can edit
+      if (!skillResource.canEdit && !auth.isAdmin()) {
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "app_auth_error",
+            message: "Only editors can modify this skill.",
+          },
+        });
+      }
+
+      // Check for existing active skill with the same name (excluding current skill)
+      const existingSkill = await SkillConfigurationModel.findOne({
+        where: {
+          workspaceId: owner.id,
+          name: body.name,
+          status: "active",
+        },
+      });
+
+      if (existingSkill && existingSkill.id !== skillResource.id) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `A skill with the name "${body.name}" already exists.`,
+          },
+        });
+      }
+
+      const updateResult = await skillResource.updateSkill(auth, {
```

Comment:
Should we run everything in a transaction to avoid leaving it in a messy state if anytging fails in the middle ?